SIPS/sip-3.md

```diff
@@ -0,0 +1,165 @@
+---
+sip: 3
+title: Transaction Insights
+status: Review
+discussions-to: https://github.com/MetaMask/SIPs/discussions/31
+author: Hassan Malik (@hmalik88)
+created: 2022-08-23
+---
+
+## Table of Contents
+
+- [Table of Contents](#table-of-contents)
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+- [Specification](#specification)
+  - [Language](#language)
+  - [Definitions](#definitions)
+  - [Common types](#common-types)
+  - [Snap Developer](#snap-developer)
+    - [Manifest](#manifest)
+    - [Snap](#snap)
+- [History](#history)
+- [Copyright](#copyright)
+
+## Abstract
+
+This SIP proposes a way for snaps to provide transaction insights that can then be displayed in the MetaMask confirmation UI.
+
+Example use-cases for transaction insights can be phishing detection, malicious address detection, transaction simulation, arbitrage trading.
+
+## Motivation
+
+One of the main use cases for transaction insights is transaction simulation. In the current state of the MetaMask wallet, we do not provide any sort of insight as to whether or not a transaction will succeed based on the web of contract/address interaction that can happen. Transaction insight snaps would provide the ability to calculate some sort of insights and return them to the wallet..
+
+## Specification
+
+> Formal specifications are written in Typescript and [JSON schema version 2020-12](https://json-schema.org/draft/2020-12/json-schema-core.html). Usage of `CAIP-N` specifications, where `N` is a number, are references to [Chain Agnostic Improvement Proposals](https://github.com/ChainAgnostic/CAIPs)
+
+### Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+"OPTIONAL" written in uppercase in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)
+
+### Definitions
+
+> This section is non-normative
+
+- `ChainId` - a [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) string. It identifies a specific chain in all of possible blockchains.
+  - `ChainId` consists of a `Namespace` and a `Reference`
+    - `Namespace` - A class of similar blockchains. For example EVM-based blockchains.
+    - `Reference` - A way to identify a concrete chain inside a `Namespace`. For example Ethereum Mainnet or Polygon.
+- `AccountId` - a [CAIP-10](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md) string. It identifies a specific account on a specific chain.
+
+### Common types
+
+The below common types are used throughout the specification
+
+````typescript
+type ChainId = string;
+
+- `ChainId` strings MUST be [CAIP-2](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-2.md) Chain Id.
+
+  The Regular Expression used to validate Chain IDs by the snap SHOULD be:
+
+  ```typescript
```

Comment:
The regex was added to SIP-2 because the wallet has to validate the data from the snap. But in case of this SIP, there shouldn't be any need for validation in usage of this API

Reply:
The `chainId` in the `onTransaction` handler function's params is CAIP-2, can't the snap developer use this regex to extract the desired chainId and use that to get whatever insight?

Reply:
Ah I see, okay that makes sense 😄 -- I'll remove it.